### PR TITLE
Fix/mushdoom movement

### PR DESCRIPTION
**Pull request by Dartemiss**
What's new?
- Several bugs fixed
- New model by tudor introduced with new animations
- RequestAttack logic changed
- Code cleaned

IMPORTANT: For the next delivery the mushdoom will be remade from scratch, but for this delivery there is no time. So the mushdoom isn't perfect at all.

### CODE REVIEW

**Comment by yalania** [comment]
some mushdooms are not doing the jump from the ground animation.

**Comment by yalania** [comment]
sometime when they are following you the animation is stuck

File: Scripts/src/Script/EnemyManager.cpp [hunk 1]
```diff
@@ -272,30 +278,21 @@ bool EnemyManager::ArePlayersInDanger() const
...
 			((Mushdoom*)enemy)->engage_player = true;
 		}
 	}
 }
 
-void EnemyManager::CancelAttack(EnemyController* enemy)
+void EnemyManager::CancelMeleeAttack(EnemyController* enemy)
 {
-	attackers.remove(enemy);
+	App->engine_log->Log("DELETING MELEE");
```
Comment (mariofv) [inline Scripts/src/Script/EnemyManager.cpp:294]: Log

File: Scripts/src/Script/EnemyManager.cpp [hunk 2]
```diff
@@ -369,3 +356,54 @@ void EnemyManager::KillAllTheEnemies()
...
 	}
 }
+
+
+void EnemyManager::NewRequestAttack(EnemyController* enemy)
+{
+	if (static_cast<Mushdoom*>(enemy)->has_ended_spawn && !IsEnemyAttacking(enemy))
+	{
+		Mushdoom* mushdoom = static_cast<Mushdoom*>(enemy);
+		App->engine_log->Log("I enter here!");
```
Comment (mariofv) [inline Scripts/src/Script/EnemyManager.cpp:366]: Log

**Review by mariofv — APPROVED** [review]
Works better, gj!

**Review by yalania — CHANGES_REQUESTED** [review]
some bugs here and there

**Review by yalania — APPROVED** [review]

**Pull request merged**